src/components/mode-toggle.tsx

```diff
@@ -12,8 +12,8 @@ export function ThemeSelect() {
     { value: "system", label: t("themeSystem") },
   ];
 
-  const handleChange = (event) => {
-    setTheme(event.target.value);
+  const handleChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setTheme(event.target.value as Theme);
```

Comment:
💅🏻  maybe, we could create a type guard for this case. Only to ensure that the value is really what we expect, instead of  using type assertion. Do you think it makes sense?

Reply:
Absolutely! I've added type guards for the value. Let me know if looks good =)